File: ask_ai.py
import os
from dotenv import load_dotenv
from openai import OpenAI

load_dotenv()

api_key = os.getenv("OPENAI_API_KEY")
if not api_key:
    raise ValueError("OPENAI_API_KEY not found in .env file")

client = OpenAI(api_key=api_key)

project_description = "платформа на основе ИИ для автоматизации поиска и анализа тендеров"

prompt = f"Придумай 3 креативных названия для моего проекта: {project_description}"

print(f"Запрос: {prompt}\n")
print("Ответ от OpenAI:")
print("-" * 40)

response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "user", "content": prompt}
    ]
)

print(response.choices[0].message.content)

File: ask_ai.mjs
import { readFileSync } from "fs";
import { fileURLToPath } from "url";
import { dirname, join } from "path";

// Read .env manually (no external deps needed)
const __dir = dirname(fileURLToPath(import.meta.url));
const envPath = join(__dir, ".env");
const envContent = readFileSync(envPath, "utf-8");
const apiKey = envContent.match(/OPENAI_API_KEY=(.+)/)?.[1]?.trim();

if (!apiKey) {
  console.error("OPENAI_API_KEY not found in .env");
  process.exit(1);
}

const projectDescription =
  "платформа на основе ИИ для автоматизации поиска и анализа тендеров";
const prompt = `Придумай 3 креативных названия для моего проекта: ${projectDescription}`;

console.log(`Запрос: ${prompt}\n`);
console.log("Ответ от OpenAI:");
console.log("-".repeat(40));

const response = await fetch("https://api.openai.com/v1/chat/completions", {
  method: "POST",
  headers: {
    "Content-Type": "application/json",
    Authorization: `Bearer ${apiKey}`,
  },
  body: JSON.stringify({
    model: "gpt-4o-mini",
    messages: [{ role: "user", content: prompt }],
  }),
});

const data = await response.json();

if (!response.ok) {
  console.error("Ошибка API:", data.error?.message ?? JSON.stringify(data));
  process.exit(1);
}

console.log(data.choices[0].message.content);
